open_event/api/events.py

```diff
@@ -122,9 +137,23 @@ def create(self, data, url):
         resource_location = url + '/' + str(new_event.id)
         return self.get(new_event.id), 201, {'Location': resource_location}
 
+    def validate(self, data, model=None):
+        # Validate Social links separately
+        social_links = data.get('social_links', [])
+
+        # Remove social links from original data
+        del data['social_links']
+        data = BaseDAO.validate(self, data, model)
+        return data, social_links
+
     def update(self, event_id, data):
-        data = self.validate(data)
+        data, social_links = self.validate(data)
         payload = self.fix_payload(data)
+        for link in social_links:
+            sl = get_object_or_404(SocialLinkModel, link['id'])
```

Comment:
Since social_links have been added to the PUT API, so it should be possible to add more social links to the event using a PUT request. But this PUT routine is not capable of that. I think it won't even allow deleting existing social links. 

I think it will be tough to properly include social links in PUT/POST API of Event (as it is a list). What about modifying the import functions to create social links using the event json. Same can be done for session types by using the sessions.json. Then we won't need to modify the API at all. (except adding length field in Session.SessionType GET API)